assets/js/googlesitekit/data/create-fetch-store.js

```diff
@@ -0,0 +1,252 @@
+ * * The fetching selector is called 'isFetchingSaveSettings'.
+ *
+ * All parts of the returned store objects should be considered internal. A
+ * public action or selector should be implemented to actually call the
+ * fetch action included in the returned store object.
+ *
+ * For example, if the fetch store is intended for an API-based
+ * action 'storeMySetting':
+ * * The basename passed should be 'storeMySetting'.
+ * * The action 'storeMySetting' should call 'fetchStoreMySetting'.
+ *
+ * Or, if the fetch store is intended for an API-based selector
+ * 'getSomeData':
+ * * The baseName passed should be 'getSomeData'.
+ * * The resolver for 'getSomeData' should call 'fetchGetSomeData'.
+ *
+ * @since n.e.x.t
+ * @private
+ *
+ * @param {Object}   options                  Options for creating the fetch store.
+ * @param {string}   options.baseName         The base name to use for all the created infrastructure.
+ * @param {Function} options.controlCallback  Callback function to issue the API request. Will be used inside the
+ *                                            control. The function receives a params object based on argsToParams,
+ *                                            i.e. the respective values passed to the action.
+ * @param {?Function} options.reducerCallback Optional. Callback function to modify state based on the API response.
+ *                                            Will be used inside the reducer. The  function receives the store's state
+ *                                            object as first parameter, the API response as second parameter, and the
+ *                                            params object for the request (see above) as third parameter. If not
+ *                                            provided, the default will return the unmodified state.
+ * @param {?Function} options.argsToParams    Optional. Function that should validate expected arguments for the
+ *                                            internal fetch action and parse them into an named parameters object,
+ *                                            with the argument names used as keys. If not provided, the default
+ *                                            function will return an empty object, essentially indicating that no
+ *                                            arguments are supported/required.
+ * @return {Object} Partial store object with properties 'actions', 'controls', 'reducer', 'resolvers', and 'selectors'.
+ */
+export const createFetchStore = ( {
+	baseName,
+	controlCallback,
+	reducerCallback,
+	argsToParams,
+} ) => {
+	invariant( baseName, 'baseName is required.' );
+	invariant( 'function' === typeof controlCallback, 'controlCallback is required.' );
+
+	if ( 'function' !== typeof reducerCallback ) {
+		reducerCallback = ( state ) => {
+			return { ...state };
+		};
+	}
+
+	let requiresParams = true;
+	if ( 'function' !== typeof argsToParams ) {
+		requiresParams = false;
```

Comment:
I've updated this to actually test the function, which is a bit more reliable than this original approach. Even when `argsToParams` is provided, it could have logic that is okay for `params` to end up empty.